templates/android/build.gradle.twig

```diff
@@ -2,17 +2,20 @@ apply plugin: 'io.github.gradle-nexus.publish-plugin'
 
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
 buildscript {
-    ext.kotlin_version = "1.8.0"
+    ext.kotlin_version = "1.9.10"
+
     version System.getenv("SDK_VERSION")
+
     repositories {
         maven { url "https://plugins.gradle.org/m2/" }
         google()
         mavenCentral()
     }
     dependencies {
-        classpath "com.android.tools.build:gradle:4.2.2"
+        classpath "com.android.tools.build:gradle:8.2.1"
         classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
-        classpath 'io.github.gradle-nexus:publish-plugin:1.1.0'
+        classpath 'io.github.gradle-nexus:publish-plugin:1.3.0'
+        classpath 'com.google.gms:google-services:4.4.0'
```

Comment:
I'm not comfortable with adding this by default, isn't there a way to avoid this and allow user to add it in their app build.gradle file instead of adding in the SDK? Doing this will complicate Flutter as we will have to add and maintain Firebase library for all Flutter platforms

Reply:
Yep removed this, it's redundant as users have to add it to their gradle build anyway. We will still need to add the Firebase BOM + messaging library as a dependency to the SDK though